client/state/themes/selectors.js

```diff
@@ -285,7 +291,12 @@ export function getThemeDetailsUrl( state, theme, siteId ) {
 		return null;
 	}
 
-	if ( isJetpackSite( state, siteId ) ) {
+	if ( isJetpackSite( state, siteId ) &&
+		! (
+			config.isEnabled( 'manage/themes/details/jetpack' ) &&
```

Comment:
Tricky, because there are three rather unrelated criteria we're checking -- the var name would need to be either something obscure like `canUseJetpack` or a concatenation of selector names and the `config` thing, and we certainly don't want that. I think it'll become easier on the eye once I'll drop the `config` flag, which I will in a subsequent PR.